tmt/frameworks/beakerlib.py

```diff
@@ -11,11 +11,22 @@
 from tmt.utils import Path
 
 if TYPE_CHECKING:
+    from tmt.base import DependencySimple, Test
     from tmt.steps.execute import TestInvocation
 
 
 @provides_framework('beakerlib')
 class Beakerlib(TestFramework):
+    @classmethod
+    def get_requirements(
+            cls,
+            test: 'Test',
+            logger: tmt.log.Logger) -> list['DependencySimple']:
+        # Avoiding circular imports: `Test.test_framework` requires `tmt.frameworks`.
+        from tmt.base import DependencySimple
+
+        return [DependencySimple('beakerlib')]
```

Comment:
Simplified in 85eb07ea. I removed the mentions of "test", I want developer to focus on framework only. Yes, requirements will be added to test requriements, just not here, it happens outside of the framework context.

Reply:
But the `test` parameter is still present, not used here, even in the follow-up #2622 it's not used. Am I missing anything?

Reply:
Yes, it's not used, but I felt it might be useful to framework one day. It's similar to why test invocation is passed other framework methods, IMHO it provides the framework methods with context. Since we don't have test invocation to use yet at this point, I felt the test is the best we can do to let the framework know the objects it's supposed to speak for. Even though no framework needs the value today.

Reply:
I see, so this is just preparation for future use. Ok.